### PR TITLE
Memory menagement HW: User space solution

### PR DESCRIPTION
**Pull request by virkha**
Add user space memory test solution

### CODE REVIEW

File: 07_memory/usr_mem_test.c [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+
+		printf("| alloc time: %ld(ns)\t", t_current.alloc_time);
+		printf("free time: %ld(ns)\n", t_current.free_time);
+	}
+
+	t_average.alloc_time = sum_alloc/MAX_SIZE;
+	t_average.free_time = sum_free/MAX_SIZE;
+
+	printf("| Average alloc time: %ld(ns)\t", t_average.alloc_time);
+	printf("average free time: %ld(ns)\n", t_average.free_time);
```
Comment (ekovalyov) [inline 07_memory/usr_mem_test.c]: Please calculation average time for each size. For implementation repeat memory allocation few times for the same size.

File: 07_memory/usr_mem_test.log [hunk 2]
```diff
@@ -0,0 +1,370 @@
+====== Run memory test allocation! ============================
+|
+| function call: malloc
+| test case:     2^x
+|
+===============================================================
+| Id: 0
+| alloc time: 1160(ns)	free time: 548(ns)
```
Comment (ekovalyov) [inline 07_memory/usr_mem_test.log]: Please print results in one line for each size.

File: 07_memory/usr_mem_test.log [hunk 3]
```diff
@@ -0,0 +1,370 @@
...
+| Id: 19
+| alloc time: 236317(ns)	free time: 0(ns)
+| Id: 20
+| alloc time: 440248(ns)	free time: 0(ns)
+| Id: 21
+| alloc time: 899967(ns)	free time: 0(ns)
+| Id: 22
+| alloc time: 1781597(ns)	free time: 0(ns)
+| Id: 23
+Segmentation fault (core dumped)
```
Comment (ekovalyov) [inline 07_memory/usr_mem_test.log:198]: Is it possible to omit this segmentation fault?

Comment (virkha) [reply 07_memory/usr_mem_test.log:198]: Yes, just add signal handler. But it's bad practice I guess.

File: 07_memory/usr_mem_test.c (hunk 1 above)
Comment (virkha) [reply 07_memory/usr_mem_test.c]: Done.

File: 07_memory/usr_mem_test.log (hunk 2 above)
Comment (virkha) [reply 07_memory/usr_mem_test.log]: Done.

**Pull request merged**